build.rs

```diff
@@ -0,0 +1,134 @@
+extern crate csv;
+extern crate serde_json;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+
+use std::ascii::AsciiExt;
+use std::io::{Write, BufReader, BufWriter};
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+use std::rc::Rc;
+use std::env;
+
+const DATA_PATH: &'static str = "misc/data.json";
+const SUPPORTED_LANG_PATH: &'static str = "misc/supported_laguages.csv";
+
+const TRIGRAPH_COUNT: usize = 300;
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+struct SupportedLang {
+    code: String,
+    eng_name: String,
+    name: String,
+    native_speakers: Option<f64>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+struct Lang<'a> {
+    info: &'a SupportedLang,
+    script: String,
+    trigraphs: Vec<String>,
+}
+
+fn main() {
+    println!("cargo:rerun-if-changed={}", DATA_PATH);
+    println!("cargo:rerun-if-changed={}", SUPPORTED_LANG_PATH);
+    let out_dir = env::var("OUT_DIR").unwrap();
+    let dest_path = Path::new(&out_dir).join("lang.rs");
```

Comment:
> At this moment I have mixed feelings: on the one hand it's a cleanup, on the other hand I'd like to keep the generated lang.rs code in the repo if it's possible. Just because it's easier to understand what's going on.

@greyblake To get that effect, the 2 lines above simply need to change to something like
```rust
let root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
let dest_path = Path::new(&out_dir).join("src").join("lang.rs");
```